### PR TITLE
Remove lodash dependency

### PR DESCRIPTION
**Pull request by simonThiele**
Hi!

This is the try to remove lodash from the tests. There were test in dev which failed already which are now passing on most test which were not checked before. So there is much to fix in the old tests first which has nothing to do with this PR.

I copied the logic from isFunction and isObject directly from lodash src to get things working.

The next steps are fixing existing tests, completely refactor qunit src since it's a bit chaotic and maybe using another testing framework like chai (is that possible or do you want to have no libs in three?)

Good start into the new week!

### CODE REVIEW

File: test/unit/SmartComparer.js [hunk 1]
```diff
@@ -68,13 +65,30 @@ function SmartComparer() {
...
+
+		return tag == '[object Function]' || tag == '[object GeneratorFunction]';
+
+  }
+
+	function isObject(value) {
+
+		// Avoid a V8 JIT bug in Chrome 19-20.
+    // See https://code.google.com/p/v8/issues/detail?id=2291 for more details.
+    var type = typeof value;
```
Comment (mrdoob) [inline test/unit/SmartComparer.js]: Indentation is inconsistent in these functions.

**Comment by mrdoob** [comment]
Looking good! What's the difference between qunit and chai?

**Comment by simonThiele** [comment]
chai has some nice features for comparing objects deeply or to check if properties exists, etc. All the stuff we need to compare things. Just an idea for the future.

File: test/unit/SmartComparer.js (hunk 1 above)
Comment (simonThiele) [inline test/unit/SmartComparer.js]: fixed

**Comment by mrdoob** [comment]
Ah, I see. Just googled Chai. Looks good! 😀

**Comment by mrdoob** [comment]
Thanks!

**Comment by simonThiele** [comment]
Thanks too!

**Comment by GGAlanSmithee** [comment]
chai is very nice!

I use it with expect to great effect.. ([example](https://github.com/GGAlanSmithee/Entities/blob/master/test/entity/manager/increase-capacity.js))

**Comment by simonThiele** [comment]
yep, we use it in our projects too with much joy :)

**Pull request merged**